tests/unit/test_adapter_yaml.py

```diff
@@ -0,0 +1,567 @@
+            none_key: null
+            bool_key: true
+            int_key: 1
+            str_key1: "value"
+            str_key2: 'value'
+            str_key3: value
+            str_multiline: |
+              value
+              value
+            complex_key:
+              complex_key2: value
+              list:
+                - item1
+                - item2
+              complex_list:
+                - item1
+                - complex_item:
+                    key: value
+            """
+        )
+        quote = quote_representation
+        fixed_source = dedent(
+            f"""\
+            ---
+            {quote}none_key{quote}:
+            {quote}bool_key{quote}: true
+            {quote}int_key{quote}: 1
+            {quote}str_key1{quote}: {quote}value{quote}
+            {quote}str_key2{quote}: {quote}value{quote}
+            {quote}str_key3{quote}: {quote}value{quote}
+            {quote}str_multiline{quote}: |
+              value
+              value
+            {quote}complex_key{quote}:
+              {quote}complex_key2{quote}: {quote}value{quote}
+              {quote}list{quote}:
+                - {quote}item1{quote}
+                - {quote}item2{quote}
+              {quote}complex_list{quote}:
+                - {quote}item1{quote}
+                - {quote}complex_item{quote}:
+                    {quote}key{quote}: {quote}value{quote}
+            """
+        )
+        config = YamlfixConfig()
+        config.quote_representation = quote_representation
+        config.quote_keys_and_basic_values = True
+
+        result = fix_code(source, config)
+
+        assert result == fixed_source
+
+    @pytest.mark.parametrize("quote_representation", quote_representations)
+    def test_quote_values_config(self, quote_representation: str) -> None:
+        """Quote only scalar values with configurable quote representation."""
+        source = dedent(
+            """\
+            none_key: null
+            bool_key: true
+            int_key: 1
+            str_key1: "value"
+            str_key2: 'value'
+            str_key3: value
+            str_multiline: |
+              value
+              value
+            complex_key:
+              complex_key2: value
+              list:
+                - item1
+                - item2
+              complex_list:
+                - item1
+                - complex_item:
+                    key: value
+            """
+        )
+        quote = quote_representation
+        fixed_source = dedent(
+            f"""\
+            ---
+            none_key:
+            bool_key: true
+            int_key: 1
+            str_key1: {quote}value{quote}
+            str_key2: {quote}value{quote}
+            str_key3: {quote}value{quote}
+            str_multiline: |
+              value
+              value
+            complex_key:
+              complex_key2: {quote}value{quote}
+              list:
+                - item1
+                - item2
```

Comment:
If we're quoting strings I'd also expect to have quoted strings in lists. In what case do you think it's interesting not to have quoted the strings in lists?